### PR TITLE
no-source build parameter handling

### PR DESCRIPTION
**Pull request by allaVolkov**
no-source build parameter handling

### Checklist
- [x] Code compiles correctly
- [x] Relevant tests were added (unit / contract / integration)
- [x] Relevant logs were added
- [x] Formatting and linting run locally successfully
- [x] All tests pass
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: internal/tpl/makefile_test.go [hunk 1]
```diff
@@ -249,4 +249,19 @@ makefile_version: 0.0.0
 		_, err := data.GetModuleDeps("m1")
 		Ω(err).Should(HaveOccurred())
 	})
+
+	It("IsNoSource fails on not existing module", func() {
```
Comment (tal-sapan) [inline internal/tpl/makefile_test.go]: Did you mean to add a test where `IsNoSource` succeeds?

File: internal/buildops/build_params_test.go [hunk 2]
```diff
@@ -440,6 +440,20 @@ var _ = Describe("GetBuilder", func() {
...
 })
 
+var _ = Describe("IfNoSource", func() {
+	It("no source module", func() {
+		buildParams := make(map[string]interface{})
+		buildParams[noSourceParam] = true
+		module := mta.Module{BuildParams: buildParams}
+		Ω(IfNoSource(&module)).Should(BeTrue())
+	})
+	It("not no source module", func() {
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/buildops/build_params_test.go]: added

File: internal/tpl/makefile_test.go (hunk 1 above)
Comment (allaVolkov) [reply internal/tpl/makefile_test.go]: no, I followed examples of GetModuleDeps that you provided

File: internal/artifacts/manifest.go [hunk 3]
```diff
@@ -122,6 +122,10 @@ func getModulesEntries(source dir.ISourceModule, targetPathGetter dir.ITargetPat
 
 	var entries []entry
 	for _, mod := range moduleList {
+		if buildops.IfNoSource(mod) {
+			continue
```
[the remark this answers is not in the document]
Comment (allaVolkov) [reply internal/artifacts/manifest.go]: fixed

**Pull request merged**